sycl/test/usm/findplatforms.hpp

```diff
@@ -0,0 +1,39 @@
+//==------------------- findplatforms.hpp ----------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+bool findPlatformAndDevice(cl_device_type deviceType,
+                           cl_platform_id &platformOut, cl_device_id &deviceOut) {
+  cl_uint numPlatforms;
+  cl_int errorCode;
+  bool foundDevice = false;
+
+  errorCode = clGetPlatformIDs(0, nullptr, &numPlatforms);
+  std::vector<cl_platform_id> platforms(numPlatforms);
+  errorCode = clGetPlatformIDs(numPlatforms, platforms.data(), nullptr);
+
+  for (auto platform : platforms) {
+    if (!foundDevice) {
+      cl_uint numDevices = 0;
+      errorCode =
+          clGetDeviceIDs(platform, deviceType, 0, nullptr, &numDevices);
+      if (errorCode != CL_SUCCESS) return false;
+
+      std::vector<cl_device_id> devices(numDevices);
+      errorCode = clGetDeviceIDs(platform, deviceType, numDevices,
+                                 devices.data(), nullptr);
+      if (errorCode != CL_SUCCESS) return false;
+
+      if (numDevices) {
+        platformOut = platform;
+        deviceOut = devices[0];
+        foundDevice = true;
```

Comment:
`foundDevice` variable seems to be also unnecessary as we are looking for one device only.
We can make code simpler if we just add `return true` here and replace all other instances of `foundDevice` with `false`.

Reply:
Yep.